sdk/keyvault/Azure.Management.KeyVault/tests/VaultOperationsTestsBase.cs

```diff
@@ -0,0 +1,194 @@
+                {
+                    if (resType.ResourceType == "vaults")
+                        return true;
+                    else
+                        return false;
+                }
+                ).First().Locations.FirstOrDefault();
+
+            rgName = Recording.GenerateAssetName("sdktestrg");
+            await ResourceGroupsClient.CreateOrUpdateAsync(rgName, new Resource.Models.ResourceGroup(location));
+            vaultName = Recording.GenerateAssetName("sdktestvault");
+
+            tenantIdGuid = new Guid(tenantId);
+            objectIdGuid = objectId;
+            tags = new Dictionary<string, string> { { "tag1", "value1" }, { "tag2", "value2" }, { "tag3", "value3" } };
+
+            var permissions = new Permissions
+            {
+                Keys = new KeyPermissions[] { new KeyPermissions("all") },
+                Secrets = new SecretPermissions[] { new SecretPermissions("all") },
+                Certificates = new CertificatePermissions[] { new CertificatePermissions("all") },
+                Storage = new StoragePermissions[] { new StoragePermissions("all") },
+            };
+            accPol = new AccessPolicyEntry(tenantIdGuid, objectId, permissions);
+
+            IList<IPRule> ipRules = new List<IPRule>();
+            ipRules.Add(new IPRule("1.2.3.4/32"));
+            ipRules.Add(new IPRule("1.0.0.0/25"));
+
+            vaultProperties = new VaultProperties(tenantIdGuid, new Sku(SkuName.Standard));
+
+
+            vaultProperties.EnabledForDeployment = true;
+            vaultProperties.EnabledForDiskEncryption = true;
+            vaultProperties.EnabledForTemplateDeployment = true;
+            vaultProperties.EnableSoftDelete = true;
+            vaultProperties.VaultUri = "";
+            vaultProperties.NetworkAcls = new NetworkRuleSet() { Bypass = "AzureServices", DefaultAction = "Allow", IpRules = ipRules, VirtualNetworkRules = null };
+            vaultProperties.AccessPolicies = new[] { accPol };
+        }
+
+        internal VaultsClient GetVaultsClient(TestRecording recording = null)
+        {
+            recording = recording ?? Recording;
+
+            return InstrumentClient(new VaultsClient(this.subscriptionId,
+                TestEnvironment.Credential,
+                recording.InstrumentClientOptions(new KeyVaultManagementClientOptions())));
+        }
+
+        internal ResourcesClient GetResourcesClient(TestRecording recording = null)
+        {
+            recording = recording ?? Recording;
+
+            return InstrumentClient(new ResourcesClient(this.subscriptionId,
+                TestEnvironment.Credential,
+                recording.InstrumentClientOptions(new ResourceManagementClientOptions())));
+        }
+
+        internal UsersClient GetGraphUsersClient(TestRecording recording = null)
+        {
+            recording = recording ?? Recording;
+
+            return InstrumentClient(new Graph.Rbac.UsersClient(this.tenantId,
```

Comment:
You don't need to instrument individual clients anymore, instrumenting the root one and calling Get***Client should work now.